app/components/Drawer.tsx

```diff
@@ -0,0 +1,73 @@
+import { Dialog, Transition } from "@headlessui/react";
+import { XMarkIcon } from "@heroicons/react/20/solid";
+import { Fragment } from "react";
+
+type DrawerProps = {
+  showDrawer: boolean;
+  setShowDrawer: (showDrawer: boolean) => void;
+};
```

Comment:
I like this interface for all modal-y things.
```suggestion
type DrawerProps = {
  open: boolean;
  onClose: () => void;
}